### ISSUE TITLE
Exception when clicking on the value of PropertyGrid's property under JAWS tool 

### ISSUE THREAD

**Issue by Amy-Li03** [dotnet/winforms#6704]
* .NET Core Version: 
.NET SDK 7.0.100-preview.2.22103

* Have you experienced this same bug with .NET Framework?: 
No

* More info: 
1. This issue reproduces on .NET 5.0 & .NET 6.0 & .NET 7.0 
2. This issue doesn't repro on .NET Core 3.1 
3. VS with other tool(e.g. Narrator) doesn't repro this issue 

**Steps to reproduce:**
[TestApp.zip](https://github.com/dotnet/winforms/files/8086107/TestApp.zip)

1. Create a .NET Core WinForms projects with any control added(E.g. Checkbox) 
2. Set the SelelctedObject property of propertyGrid to the checkBox control 
3. Run this project 
4. Launch the **JAWS** tool 
5. Clicking using mouse or press "Tab" keyboard on the value of propertyGrid's property 
![Actual](https://user-images.githubusercontent.com/62929087/154419615-f42458cc-c068-4610-94a3-61cfb5235bf8.gif)

**Expected behavior:**
No exception was throwing.

**Actual behavior:**
There is an exception was throwing 
![image](https://user-images.githubusercontent.com/62929087/154417935-4640850f-1776-4c47-9354-05da7615e447.png)

**Call Stack:**
System.ExecutionEngineException
  HResult=0x80131506
  Source= Cannot evaluate the exception source
  StackTrace: Cannot evaluate the exception stack trace

**Comment by RussKie** [comment]
@merriemcgaw do we need to service this?

**Comment by merriemcgaw** [comment]
@SergeySmirnov-Akvelon @Tanya-Solyanik did this bug get in 6.0? If so let's fix it in just 6.0, nothing lower.

**Comment by Ashley-Li** [comment]
Verified this issue in .Net 6.0.202 service build, it is fixed that no exception was throwing when clicking using mouse or press "Tab" keyboard on the value of propertyGrid's property. Please see below test result:
![result](https://user-images.githubusercontent.com/56664234/160409170-ff7c3494-bba9-469b-969d-819d08158efe.gif)

**Comment by John-Qiao** [comment]
Verified with .NET SDK 7.0.100-preview.4.22201.8 build, this issue is fixed as same as above test result.

**Comment by John-Qiao** [comment]
Verified with .NET SDK 7.0.100-preview.3.22179.4 build, this issue is fixed as same as above test result.
